src/Propel/Generator/Reverse/PgsqlSchemaParser.php

```diff
@@ -417,7 +417,10 @@ protected function addForeignKeys(Table $table, $oid)
                 $fk = new ForeignKey($name);
                 $fk->setForeignTableCommonName($foreignTable->getCommonName());
                 if ($table->guessSchemaName() != $foreignTable->guessSchemaName()) {
-                    $fk->setForeignSchemaName($foreignTable->getSchema());
+                     if($foreignTable->getSchema())
+                            $fk->setForeignSchemaName($foreignTable->getSchema());
+                     elseif($foreignTable->getSchema() == 'public')
```

Comment:
That is weird. It should be set via the first `setForeignSchema()` call. If that is not the case, then the bug hides in that method. Calling however the same method twice with the same arguments directly after each other is really not a bugfix, and it is currently not logical at all. I just don't see how this fixes anything. Maybe you can provide a unit-test that proves that this codes fixes it.